src/python/pants/backend/cc/goals/tailor_test.py

```diff
@@ -38,7 +38,9 @@ def test_find_putative_targets(rule_runner: RuleRunner) -> None:
     putative_targets = rule_runner.request(
         PutativeTargets,
         [
-            PutativeCCTargetsRequest(PutativeTargetsSearchPaths(("",))),
+            PutativeCCTargetsRequest(
+                PutativeTargetsSearchPaths(("src/native/owned", "src/native/unowned"))
+            ),
```

Comment:
Ack this new test boilerplate isn't the best. Please keep reading on to `core/goals/tailor.py` for why I implemented this way.

Reply:
Given that the semantics of this class have changed anyway (is now expected to be non-recursive), it seems like it could also be... removed? There doesn't seem to be a good reason for the split between `PutativeTargetsRequest` and `PutativeTargetsSearchPaths`.

Reply:
Yeah true. I'm not sure why we have a dedicated object for this when it could live directly on the request object. I'll fix, but I think better as a dedicated change so I can mark it "Plugin API change"